### ISSUE TITLE
False positive "Match must be exhaustive" when using #[non_exhaustive] in the same crate

### ISSUE THREAD

**Issue by Mcat12** [intellij-rust/intellij-rust#4896]
## Environment

* **IntelliJ Rust plugin version:** 0.2.114.2151-193
* **Rust toolchain version:** 1.40.0
* **IDE name and version:** CLion 2019.3.3
* **Operating system:** Linux (Fedora 31)

## Problem description
`#[non_exhaustive]` was introduced in [Rust 1.40.0](https://blog.rust-lang.org/2019/12/19/Rust-1.40.0.html#[non_exhaustive]-structs,-enums,-and-variants). It prevents users of a crate from exhaustively matching on whatever is annotated with `#[non_exhaustive]`. However, that restriction does not apply to the crate in which the type is defined. [Link to section in RFC which mentions this](https://github.com/rust-lang/rfcs/blob/master/text/2008-non-exhaustive.md#enums-1)

IntelliJ Rust correctly identifies that matching on a `#[non_exhaustive]` type is prohibited, but it continues to throw that error when matching on the type inside the declaring crate.

## Steps to reproduce

```rust
#[non_exhaustive]
enum Error {
    Variant
}

fn main() {
    let error = Error::Variant;

    match error {
        Error::Variant => println!("Variant")
    }
}
```

![image](https://user-images.githubusercontent.com/4417660/73321180-ab0e3880-420f-11ea-8c96-54f3cde8fba6.png)
